core/src/database/verifier.rs

```diff
@@ -37,4 +40,245 @@ impl Database {
 
         Ok(pks.into_iter().map(|(_, pk)| pk.0).collect())
     }
+
+    pub async fn set_move_to_vault_txid_from_citrea_deposit(
+        &self,
+        tx: Option<DatabaseTransaction<'_, '_>>,
+        idx: u32,
```

Comment:
what idx is this? it'd be better to name it. if it's a unique database id, then id would be better. if it's tied to citrea, citrea_idx would be better